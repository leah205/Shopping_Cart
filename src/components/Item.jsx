import styled from "styled-components"
import {useState} from "react"

const ItemImg = styled.img`
   
    margin: auto;
    display: block;
    width: 100px;
`

const StyledNumberBtn = styled.button`
    background: ${props => props.$decrement? "red" : "green"};
    border: 0;
    width: 30px;
    height: 30px;
    color: white;
`
const ItemCard = styled.div`
    width: 200px;
    border: 1px solid gray;
    border-radius: 5px;
    box-shadow: 3px 5px 5px gray;
    padding: 10px;
`
const SelectAmountSection = styled.div`
    display: flex;
    gap: 5px;
    margin-bottom: 10px;
    `
const ItemInput = styled.input`
    width: 100px;
`

function ItemNumberBtn({children, $decrement, updateCount}){
    const handleClick = () => {
         if($decrement){
            updateCount('decrement')
        } else {
            updateCount('increment')
        }
    }
    return <StyledNumberBtn onClick = {handleClick} $decrement = {$decrement} >{children}</StyledNumberBtn>
}


export default function Item({id, title, price, description, image, rating, addItemsToCart}){ 
    const [countValue, setCountValue] = useState(0)
    const handleCountClick = (type) => {
        if(type == 'decrement'){
            if (countValue == 0){
                return
            }
            setCountValue(countValue - 1)
        } else{
            setCountValue(countValue + 1)
        }
    }

    function clickAddToCart(){
        let details = {title: title, count: countValue, image:image, price: price, id: id}
        addItemsToCart(details)
        setCountValue(0)
    }
    
    return <>
        <ItemCard>
            <ItemImg src = {image}></ItemImg>
            <h2>{title}</h2>
            <p>{price}</p>
            <p>{description}</p>
           <p>{rating.rate} by {rating.count}</p>
            <SelectAmountSection>
                <ItemNumberBtn $decrement = {true} count = {countValue} updateCount = {handleCountClick}>-</ItemNumberBtn>
                <ItemInput type = "number" 
                value = {countValue} 
                onChange = {(event) => setCountValue(event.target.value)}></ItemInput>
                <ItemNumberBtn $increment = {true} count = {countValue} updateCount = {handleCountClick}>+</ItemNumberBtn>
                
            </SelectAmountSection>
            
            
            <button onClick = {clickAddToCart}>Add to Cart</button>
        </ItemCard>
    </>
}