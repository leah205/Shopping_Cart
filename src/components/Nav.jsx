import React from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";
import cartSrc from "../assets/cart.svg"
import PropTypes from "prop-types";

const StyledNav = styled.nav`
    display: flex;
    list-style-type: none;
    text-decoration: none;
    width: 100vw;
    background: navy;
    align-items: center;
    padding: 0px 15px;
`

const NavList = styled.ul`
    list-style-type: none;
    display: flex;
    justify-content:space-between;
    width:100%;
    align-items: center;
    div{
    display:flex;
    align-items: center;

    }
    
`
const StyledLink = styled.li`
    &:hover{
    background: rgb(149, 149, 178);;
}
    height: 100%;
    padding: 5px;
    color: rgb(173, 173, 255);
`


const StyledCartButton = styled.div`
    background: none;
    border: 0;

    img{
    height: 40px}
`

function CartButton({ItemCount, handleClick}){
   console.log(ItemCount)
    return <StyledCartButton onClick = {() => handleClick(true)}>
        <p>{ItemCount}</p>
        <img src= {cartSrc} alt="cart" />
    </StyledCartButton>
}

CartButton.propTypes = {
    ItemCount: PropTypes.number,
    handleClick: PropTypes.func
}



export default function Nav({itemCount,  setShoppingModal}){
    console.log(itemCount)
    return (
        <StyledNav>
            <NavList>
                <div>
                <StyledLink><Link to = "home">Home</Link></StyledLink>
                <StyledLink><Link to = "shop">Shop </Link></StyledLink>
                </div>
                <StyledLink><CartButton 
                ItemCount = {itemCount}
                handleClick = {setShoppingModal}
                ></CartButton></StyledLink>

            </NavList>
            
        </StyledNav> )
}

Nav.propTypes = {
    itemCount: PropTypes.number,
    setShoppingModal: PropTypes.func
}

