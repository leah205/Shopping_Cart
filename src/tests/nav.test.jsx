import React from "react";
import {render, screen} from '@testing-library/react'
import { describe, expect, it, vi } from "vitest";
import Nav from "../components/Nav"
import Item from "../components/Item"
import {BrowserRouter} from 'react-router-dom';
const noop = () => {}
describe("nav", ()=> {
    it("renders properly", () => {
    
         let nav = render(
            <BrowserRouter> <Nav></Nav></BrowserRouter>
        
        )
         expect(nav).toMatchSnapshot()
    });
    it("item count displays next to cart button", () => {
        render(<BrowserRouter><Nav itemCount = {3} setShoppingModal = {noop}></Nav></BrowserRouter>)
        const itemCount = screen.getByTestId("item-count")
        expect(itemCount.textContent).toBe("3")
    })
    
})