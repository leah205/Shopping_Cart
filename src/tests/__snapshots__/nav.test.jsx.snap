// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nav > renders properly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <nav
        class="sc-bRKDuR eapIse"
      >
        <ul
          class="sc-hvigdm ZqINI"
        >
          <div>
            <li
              class="sc-fhHczv dwvuCB"
            >
              <a
                data-discover="true"
                href="/home"
              >
                Home
              </a>
            </li>
            <li
              class="sc-fhHczv dwvuCB"
            >
              <a
                data-discover="true"
                href="/shop"
              >
                Shop 
              </a>
            </li>
          </div>
          <li
            class="sc-fhHczv dwvuCB"
          >
            <div
              class="sc-ggWZvA bpLcjf"
            >
              <p
                data-testid="item-count"
              />
              <img
                alt="cart"
                src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='utf-8'?%3e%3c!--%20Uploaded%20to:%20SVG%20Repo,%20www.svgrepo.com,%20Generator:%20SVG%20Repo%20Mixer%20Tools%20--%3e%3csvg%20fill%20=%20'white'%20width='800px'%20height='800px'%20viewBox='0%200%2024%2024'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20fill%20=%20'none'%20d='M6.29977%205H21L19%2012H7.37671M20%2016H8L6%203H3M9%2020C9%2020.5523%208.55228%2021%208%2021C7.44772%2021%207%2020.5523%207%2020C7%2019.4477%207.44772%2019%208%2019C8.55228%2019%209%2019.4477%209%2020ZM20%2020C20%2020.5523%2019.5523%2021%2019%2021C18.4477%2021%2018%2020.5523%2018%2020C18%2019.4477%2018.4477%2019%2019%2019C19.5523%2019%2020%2019.4477%2020%2020Z'%20stroke='white'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
              />
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
     
    <nav
      class="sc-bRKDuR eapIse"
    >
      <ul
        class="sc-hvigdm ZqINI"
      >
        <div>
          <li
            class="sc-fhHczv dwvuCB"
          >
            <a
              data-discover="true"
              href="/home"
            >
              Home
            </a>
          </li>
          <li
            class="sc-fhHczv dwvuCB"
          >
            <a
              data-discover="true"
              href="/shop"
            >
              Shop 
            </a>
          </li>
        </div>
        <li
          class="sc-fhHczv dwvuCB"
        >
          <div
            class="sc-ggWZvA bpLcjf"
          >
            <p
              data-testid="item-count"
            />
            <img
              alt="cart"
              src="data:image/svg+xml,%3c?xml%20version='1.0'%20encoding='utf-8'?%3e%3c!--%20Uploaded%20to:%20SVG%20Repo,%20www.svgrepo.com,%20Generator:%20SVG%20Repo%20Mixer%20Tools%20--%3e%3csvg%20fill%20=%20'white'%20width='800px'%20height='800px'%20viewBox='0%200%2024%2024'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20fill%20=%20'none'%20d='M6.29977%205H21L19%2012H7.37671M20%2016H8L6%203H3M9%2020C9%2020.5523%208.55228%2021%208%2021C7.44772%2021%207%2020.5523%207%2020C7%2019.4477%207.44772%2019%208%2019C8.55228%2019%209%2019.4477%209%2020ZM20%2020C20%2020.5523%2019.5523%2021%2019%2021C18.4477%2021%2018%2020.5523%2018%2020C18%2019.4477%2018.4477%2019%2019%2019C19.5523%2019%2020%2019.4477%2020%2020Z'%20stroke='white'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
            />
          </div>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
